iiify/search.py

```diff
@@ -0,0 +1,69 @@
+import requests
+from .resolver import ARCHIVE, URI_PRIFIX
+import json
+import re
+
+def buildSearchURL(identifier, query):
+    response = requests.get(f"{ARCHIVE}/metadata/{identifier}")
+    response.raise_for_status() 
+
+    metadata = response.json()
+
+    return f"https://{metadata['server']}/fulltext/inside.php?item_id={identifier}&doc={identifier}&path={metadata['dir']}&q={query}"
+
+def iiif_search(identifier, query):
+    url = buildSearchURL(identifier, query)
+
+    print (f"Search URL:\n{url}")
+    response = requests.get(url)
+    response.raise_for_status() 
+    ia_response = response.json()
+
+    searchResponse = {
+        "@context":"http://iiif.io/api/presentation/2/context.json",
+        "@id": f"{URI_PRIFIX}/search/{identifier}?q={query}",
+        "@type":"sc:AnnotationList",
+
+        "resources": [
+        ]
+    }
+
+    # "b": 1090, "t": 700, "r": 2024, "l": 192, 
+    count = 1
+    for match in ia_response['matches']:
+        paragraph = match['par'][0]
+       
+        # Only show the match rather than the full matching paragraph 
+        text = ""
+        largeResults = False
+        if len(match['text']) < 10000:
+            largeResults = True
+            match = re.findall(r"<IA_FTS_MATCH>(.*)</IA_FTS_MATCH>", match['text'])
```

Comment:
As I said in my comment, I think we should use bs4 for this instead of `re`.  Before we do that though, does it make sense to break this into a smaller function so we can have a unit test that proves that the `re` / `bs4` approaches are the same but with CodeQL happy?

Reply:
As discussed the testing counts the number of matches so will break if the `bs4` wasn't working.